### PR TITLE
Add failure_reason to nexus_task_execution_failed

### PR DESCRIPTION
**Pull request by Quinn-With-Two-Ns**
Add failure_reason to nexus_task_execution_failed based on the Go SDK PR https://github.com/temporalio/sdk-go/pull/1671

closes https://github.com/temporalio/sdk-java/issues/2273

### CODE REVIEW

**Comment by cretz** [comment]
Would like to get confirm the expected enumerates on the other PR before merging this one.

**Comment by Quinn-With-Two-Ns** [comment]
@cretz good to merge now that Go is merged?

**Review by cretz — APPROVED** [review]
LGTM. I have been told by @bergundy that it's `handler_error_<TYPE>` where `TYPE` is SCREAMING_SNAKE_CASE, so I assume the same here.

File: temporal-sdk/src/test/java/io/temporal/workflow/nexus/OperationFailMetricTest.java [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+import io.temporal.workflow.shared.TestNexusServices;
+import io.temporal.workflow.shared.TestWorkflows.TestWorkflow1;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class OperationFailMetricTest {
```
Comment (Quinn-With-Two-Ns) [inline temporal-sdk/src/test/java/io/temporal/workflow/nexus/OperationFailMetricTest.java:52]: @cretz these tests should hopefully assert parity between the Go and Java SDK

**Pull request merged**